### PR TITLE
Fix tensorrt subgraph bug

### LINKED ISSUE #13120: Fix tensorrt subgraph bug.
Suppose we have a graph G.  The G can be describe as follows:
`A -> B -> C,  A -> D -> C`.    There are 4 nodes in the graph. We suppose that `A, B, C `can be convert to tensorrt nodes.  According to the algo we support now, A, B, C will be merged into a subgraph SubG. So, the result graph will be` SubG -> D, D-> SubG`, there will be a cycle here. 

To prevent the occurrence of the above situation, we must guarantee that if node x and node y in the subgraph then all path from x and y are in the subgraph.

### PR DESCRIPTION
**Pull request by NHZlX**
fix #13120

### CODE REVIEW

File: paddle/fluid/inference/analysis/subgraph_splitter.cc [hunk 1]
```diff
@@ -74,25 +74,200 @@ void UnionFindCombine(const node_map_t &node_map, size_t a, size_t b) {
...
+// This is to avoid changing the original graph
+// in the process of trt graph analysis.
+struct BriefNode {
+  explicit BriefNode(Node *n) { node = n; }
+  Node *node;
+  std::vector<BriefNode *> inlinks;
+  std::vector<BriefNode *> outlinks;
+};
+
+void UnionContractedNodes(const std::unordered_map<int, BriefNode *> &node_map,
```
Comment (Superjomn) [inline paddle/fluid/inference/analysis/subgraph_splitter.cc]: some comments here?

File: paddle/fluid/operators/tensorrt_engine_op.h [hunk 2]
```diff
@@ -160,11 +160,21 @@ class TensorRTEngineKernel : public framework::OpKernel<T> {
           fluid_t->mutable_data<float>(platform::CUDAPlace(
               boost::get<platform::CUDAPlace>(context.GetPlace()).device)),
           size * sizeof(float));
-      //} else {
-      // engine->GetOutputInGPU(
-      // y, fluid_t->mutable_data<float>(platform::CUDAPlace()),
-      // size * sizeof(float));
-      //}
+
+      // TODO(zhaolong) : delete it sometimes
```
Comment (Superjomn) [inline paddle/fluid/operators/tensorrt_engine_op.h]: remove these

File: paddle/fluid/inference/tensorrt/convert/conv2d_op.cc [hunk 3]
```diff
@@ -26,6 +26,9 @@ class Conv2dOpConverter : public OpConverter {
         << "convert a fluid conv2d op to tensorrt conv layer without bias";
 
     framework::OpDesc op_desc(op, nullptr);
+    std::cout << "Conv op: " << std::endl;
```
Comment (Superjomn) [inline paddle/fluid/inference/tensorrt/convert/conv2d_op.cc]: Turn these `cout` to `VLOG(4)` ?

File: paddle/fluid/inference/tensorrt/convert/concat_op.cc [hunk 4]
```diff
@@ -30,7 +30,9 @@ class ConcatOpConverter : public OpConverter {
     framework::OpDesc op_desc(op, nullptr);
     // Declare inputs
     std::vector<nvinfer1::ITensor*> itensors;
+    std::cout << "Concat op: " << std::endl;
```
Comment (Superjomn) [inline paddle/fluid/inference/tensorrt/convert/concat_op.cc]: make these `cout` to VLOG

File: paddle/fluid/inference/tensorrt/convert/conv2d_op.cc (hunk 3 above)
Comment (NHZlX) [reply paddle/fluid/inference/tensorrt/convert/conv2d_op.cc]: I forget to delete these unused log.

File: paddle/fluid/inference/tensorrt/convert/concat_op.cc (hunk 4 above)
Comment (NHZlX) [reply paddle/fluid/inference/tensorrt/convert/concat_op.cc]: Done!

File: paddle/fluid/inference/analysis/subgraph_splitter.cc (hunk 1 above)
Comment (NHZlX) [reply paddle/fluid/inference/analysis/subgraph_splitter.cc]: Done!

File: paddle/fluid/operators/tensorrt_engine_op.h (hunk 2 above)
Comment (NHZlX) [reply paddle/fluid/operators/tensorrt_engine_op.h]: Done!

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**